Avoid geometry update assertions in some tests when UI side compositing is enabled on macOS

heycam opened the pull request:
#### 216acdc0bca8c93ccd03e1fd0706bcbe5270652b
<pre>
Avoid geometry update assertions in some tests when UI side compositing is enabled on macOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=242884">https://bugs.webkit.org/show_bug.cgi?id=242884</a>
rdar://97583976

Reviewed by Kimmo Kinnunen.

Some tests hit this assertion when run on macOS with UI side compositing
enabeld:

ASSERTION FAILED: m_isWaitingForDidUpdateGeometry
.../OpenSource/Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm(99) : virtual void WebKit::RemoteLayerTreeDrawingAreaProxy::didUpdateGeometry()

This is due to the set calling testRunner.setViewSize(...), which causes
UpdateGeometry / DidUpdateGeometry messages to be sent synchronously.

This means we re-entrantly call RemoteLayerTreeDrawingAreaProxy::didUpdateGeometry,
which asserts m_isWaitingForDidUpdateGeometry, before the
`m_isWaitingForDidUpdateGeometry = true` assignment just after the
send(Messages::DrawingArea::UpdateGeometry(...)) call in
RemoteLayerTreeDrawingAreaProxy::sendUpdateGeometry is run.

Move the assignment before the send() call to avoid this.

* LayoutTests/platform/ios-simulator-wk2/accessibility/visible-character-range-height-changes-expected.txt: Added.
* LayoutTests/platform/ios-simulator-wk2/accessibility/visible-character-range-width-changes-expected.txt: Added.
* LayoutTests/platform/ios-simulator-wk2/TestExpectations:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::sendUpdateGeometry):

Canonical link: <a href="https://commits.webkit.org/256042@main">https://commits.webkit.org/256042@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/182930dc3c39ac739571d37646b9858282bafa76

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94395 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3574 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27314 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104056 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164333 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3623 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31776 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86730 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100052 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100065 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2598 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80790 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29637 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84520 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/84114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/72528 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38174 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/17938 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36046 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/19211 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39935 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41838 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1982 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41888 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/38440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple reviewed (approved):
It's a message with an async reply

heycam committed:
Avoid geometry update assertions in some tests when UI side compositing is enabled on macOS
https://bugs.webkit.org/show_bug.cgi?id=242884
rdar://97583976

Reviewed by Kimmo Kinnunen.

Some tests hit this assertion when run on macOS with UI side compositing
enabeld:

ASSERTION FAILED: m_isWaitingForDidUpdateGeometry
.../OpenSource/Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm(99) : virtual void WebKit::RemoteLayerTreeDrawingAreaProxy::didUpdateGeometry()

This is due to the set calling testRunner.setViewSize(...), which causes
UpdateGeometry / DidUpdateGeometry messages to be sent synchronously.

This means we re-entrantly call RemoteLayerTreeDrawingAreaProxy::didUpdateGeometry,
which asserts m_isWaitingForDidUpdateGeometry, before the
`m_isWaitingForDidUpdateGeometry = true` assignment just after the
send(Messages::DrawingArea::UpdateGeometry(...)) call in
RemoteLayerTreeDrawingAreaProxy::sendUpdateGeometry is run.

Move the assignment before the send() call to avoid this.

* LayoutTests/platform/ios-simulator-wk2/accessibility/visible-character-range-height-changes-expected.txt: Added.
* LayoutTests/platform/ios-simulator-wk2/accessibility/visible-character-range-width-changes-expected.txt: Added.
* LayoutTests/platform/ios-simulator-wk2/TestExpectations:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::sendUpdateGeometry):

Canonical link: https://commits.webkit.org/256042@main
